projects/igniteui-angular/src/lib/grids/api.service.ts

```diff
@@ -326,6 +326,9 @@ export class GridBaseAPIService <T extends IgxGridBaseComponent & IGridDataBinda
         }
 
         grid.filteringExpressionsTree = filteringTree;
+        if (!condition) {
```

Comment:
You could move this check at the beginning of the method. However I think that for filterGlobal, we should make the condition argument required. This method doesn't make sense to be called without condition. Both changes are breaking, but I think that compilation error is better than a runtime error in this case.